src/services/upower/core.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <qdbusservicewatcher.h>
+#include <qhash.h>
+#include <qobject.h>
+#include <qqmlintegration.h>
+#include <qqmllist.h>
+#include <qtmetamacros.h>
+
+#include "../../core/model.hpp"
+#include "dbus_app.h"
+#include "device.hpp"
+
+namespace qs::service::upower {
+
+class UPower: public QObject {
+	Q_OBJECT;
+	QML_NAMED_ELEMENT(UPower);
+	QML_SINGLETON;
```

Comment:
Instead of using a singleton, use a global instance and a singleton as a proxy object as is done in [mpris](https://github.com/outfoxxed/quickshell/blob/master/src/services/mpris/watcher.hpp). This allows known state to persist across reloads instead of re-querying it every time.